module/config.mjs

```diff
@@ -1897,6 +1897,170 @@ DND5E.advancementTypes = {
 
 /* -------------------------------------------- */
 
+/**
+ * Status effects to display on the token HUD. Each internal object contains name, description,
+ * icon, and optionally changes and additional status ids that should apply. For example, if an actor
+ * is petrified, they are also incapacitated and silenced.
+ * @type {object}
+ */
+DND5E.conditionEffects = {
+  dead: {
+    name: "DND5E.ConDead",
+    description: "DND5E.ConDeadText",
+    icon: "icons/svg/skull.svg"
+  },
+  blinded: {
+    name: "DND5E.ConBlinded",
+    description: "DND5E.ConBlindedText",
+    icon: "icons/svg/blind.svg"
+  },
+  bleeding: {
+    name: "DND5E.ConBleeding",
+    description: "DND5E.ConBleedingText",
+    icon: "icons/svg/blood.svg"
+  },
+  charmed: {
+    name: "DND5E.ConCharmed",
+    description: "DND5E.ConCharmedText",
+    icon: "icons/svg/heal.svg"
+  },
+  cursed: {
+    name: "DND5E.ConCursed",
+    description: "DND5E.ConCursedText",
+    icon: "icons/svg/sun.svg"
+  },
+  deafened: {
+    name: "DND5E.ConDeafened",
+    description: "DND5E.ConDeafenedText",
+    icon: "icons/svg/deaf.svg"
+  },
+  diseased: {
+    name: "DND5E.ConDiseased",
+    description: "DND5E.ConDiseasedText",
+    icon: "icons/svg/biohazard.svg"
+  },
+  frightened: {
+    name: "DND5E.ConFrightened",
+    description: "DND5E.ConFrightenedText",
+    icon: "icons/svg/terror.svg"
+  },
+  grappled: {
```

Comment:
I have one key concern here that I'd hope to discuss before moving forward regarding directionality and control.

Does the `ActiveEffect` include a condition? Or does the condition generate the Active Effect? I prefer the first path, but the data structures in this PR imply the second path.

The design that core Foundry now has is that ActiveEffect embedded documents each contain a set of `statuses`, and those respective status sets can be unioned to define a set of `statuses` which currently apply to the Actor.

This allows us to encapsulate more nuanced logic inside methods like `#prepareMovementSpeed`. For example consider a creature that has an ActiveEffect that imparts the `grappled` status, but also has condition immunity to `grappled`. This approach where the condition generates a certain active effect breaks down, whereas a scenario where we prepare actor data conditional on the actor's current statuses allows us to automate those sorts of immunities or interactions.

I think this is very important to shift, and I may not have explained myself fully or clearly so I hope we can discuss.

Reply:
> I may not have explained myself fully or clearly so I hope we can discuss.

What I am understanding is that the conditions here should not have an array of `changes`.

That's all easily amended, though I'm interested in seeing how we will handle the 'changes' inside data preparation given that `CONFIG.DND5E.conditionTypes` is configurable.